### PR TITLE
Suggestion: Add CLI to override config values

### PR DESCRIPTION
**Pull request by regadas**
Hi @fthomas I sketched this really quick and it might be a bit rough, let me know what you think!

### CODE REVIEW

File: modules/core/src/main/scala/eu/timepit/scalasteward/application/Cli.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      reposDir: Option[String],
+      reposFile: String,
+      gitAuthorName: String,
+      gitAuthorEmail: String,
+      githubApiHost: String,
+      githubLogin: String,
+      gitAskPass: String
+  )
+
+  def create[F[_]](implicit F: Sync[F]): Cli[F] = new Cli[F] {
```
Comment (fthomas) [inline modules/core/src/main/scala/eu/timepit/scalasteward/application/Cli.scala]: It seems to me that `ApplicativeError[F, Throwable]` would be enough here. There is also a handy `ApplicativeThrowable` for the former in the `util` package object

File: modules/core/src/main/scala/eu/timepit/scalasteward/application/Config.scala [hunk 2]
```diff
@@ -45,6 +45,8 @@ import scala.sys.process.Process
   */
 final case class Config(
     workspace: File,
+    reposFile: File,
+    reposDir: File,
```
Comment (fthomas) [inline modules/core/src/main/scala/eu/timepit/scalasteward/application/Config.scala]: What is the reason for adding `reposDir`? Why would you want to override the default `workspace / "repos"`?

File: modules/core/src/main/scala/eu/timepit/scalasteward/application/Cli.scala (hunk 1 above)
Comment (regadas) [reply modules/core/src/main/scala/eu/timepit/scalasteward/application/Cli.scala]: Good point!

File: modules/core/src/main/scala/eu/timepit/scalasteward/application/Config.scala (hunk 2 above)
Comment (regadas) [reply modules/core/src/main/scala/eu/timepit/scalasteward/application/Config.scala]: yeah, on a second thought I guess it doesn't make that much sense now 😄

**Pull request merged**